heron/api/src/scala/com/twitter/heron/streamlet/KeyedWindow.scala

```diff
@@ -0,0 +1,47 @@
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//  Copyright 2017 Twitter. All rights reserved.
+package com.twitter.heron.streamlet.scala
+
+import java.io.Serializable
+
+import scala.beans.{BeanProperty, BooleanBeanProperty}
+
+//remove if not needed
+import scala.collection.JavaConversions._
+
+/**
+  * Transformation depending on Windowing pass on the window/key information
+  * using this class
+  */
+@SerialVersionUID(4193319775040181971L)
+class KeyedWindow[T]() extends Serializable {
+
+  @BeanProperty
+  var key: T = _
+
+  @BeanProperty
+  var window: Window = _
+
+  def this(key: T, window: Window) = {
+    this()
+    this.key = key
+    this.window = window
+  }
+
+  override def toString(): String =
+    "{ Key: " + String.valueOf(key) + " Window: " + window.toString +
```

Comment:
do we really need `String.valueOf` or `toString`? if you just `"{ key: " + key + " window: " + window + " }"`, it is better than explicitly call `toString`, because it handles null field, while `window.toString` would throw NPE if `window` is null.